lib/socketry/exceptions.rb

```diff
@@ -0,0 +1,18 @@
+module Socketry
+  # Generic catch all for all Socketry errors
+  SocketError = Class.new(IOError)
```

Comment:
- why not simply `Error`? IMHO `Socketry::Error` looks better than `Socketry::SocketError`.
- why `MyClass = Class.new(ParentClass)`? Why not `class MyClass < ParentClass; end`?